File: app/src/main/java/com/mahmoudmabrok/hadibadi/Datalayer/Local/Sharedmanager.java
package com.mahmoudmabrok.hadibadi.Datalayer.Local;

import android.content.Context;
import android.content.SharedPreferences;

import java.util.ArrayList;
import java.util.List;
import java.util.Set;
import java.util.TreeSet;

/**
 * Created by Mahmoud on 9/11/2018.
 */
public class Sharedmanager {
    private SharedPreferences sharedPreferences;
    private SharedPreferences.Editor editor;

    public Sharedmanager(Context context) {
        sharedPreferences = context.getSharedPreferences("teams", Context.MODE_PRIVATE);
        editor = sharedPreferences.edit();
    }

    public void setTeams(List<String> teams) {
        Set<String> set = new TreeSet<>();
        set.addAll(teams);
        editor.putStringSet("teams", set);
    }

    public List<String> getTeamList() {
        List<String> list = new ArrayList<>();
        try {
            list.addAll(sharedPreferences.getStringSet("teams", null));
        } catch (Exception e) {
            e.printStackTrace();
        }
        return list;
    }


}

File: app/src/main/java/com/mahmoudmabrok/hadibadi/Datalayer/Local/Sharedmanager.kt
package com.mahmoudmabrok.hadibadi.Datalayer.Local

import android.content.Context
import android.content.SharedPreferences
import java.util.*

/**
 * Created by Mahmoud on 9/11/2018.
 */
class Sharedmanager(context: Context) {
    private val sharedPreferences: SharedPreferences = context.getSharedPreferences("teams", Context.MODE_PRIVATE)
    private val editor: SharedPreferences.Editor

    fun setTeams(teams: List<String>, name: String) {
        val set: MutableSet<String> = TreeSet()
        set.addAll(teams)
        editor.putStringSet(name, set)
    }

    fun getTeams(name: String): List<String> {
        val list = mutableListOf<String>()
        val temasSet = sharedPreferences.getStringSet(name, null)
        temasSet?.let { list.addAll(it.toList()) }
        return list
    }

    init {
        editor = sharedPreferences.edit()
    }
}